app/src/main/java/de/tum/in/tumcampusapp/adapters/ChatHistoryAdapter.java

```diff
@@ -47,146 +43,141 @@ public void removeUnsent(ChatMessage msg) {
-            mCursor.moveToPosition(position);
-            return mCursor.getLong(ChatMessageManager.COL_ID);
+            return chatHistoryList.get(position).getId();
         } else {
             return 0;
         }
     }
 
-    @Override
+
     public int getViewTypeCount() {
         return 2;
     }
 
     @Override
     public int getItemViewType(int position) {
-        if (position > super.getCount()) {
+        int count = getSentCount();
+        if (position > count) {
             return 0;
         }
         ChatMessage msg = (ChatMessage) getItem(position);
         return currentChatMember.getId() == msg.getMember()
                                                .getId() ? 0 : 1;
     }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup viewGroup) {
-        int count = super.getCount();
-        if (position < count) {
-            return super.getView(position, convertView, viewGroup);
-        }
-
-        ChatMessage chatMessage = unsentMessages.get(position - count);
-        View v = newView(mContext, null, viewGroup);
-        bindViewChatMessage(v, chatMessage);
-        return v;
-    }
 
     @Override
-    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
-        ViewHolder holder;
-        boolean outgoing = true;
-        if (cursor != null) {
-            ChatMessage msg = ChatMessageManager.toObject(cursor);
-            outgoing = currentChatMember.getId() == msg.getMember()
-                                                       .getId();
+    public View getView(int position, View convertView, ViewGroup parent)   {
+        boolean outgoing;
+
+        if (getItemViewType(position)==0)   {
+            outgoing = true;
+        } else {
+            outgoing = false;
         }
 
         int layout = outgoing ? R.layout.activity_chat_history_row_outgoing : R.layout.activity_chat_history_row_incoming;
-        View view = inflater.inflate(layout, viewGroup, false);
-        holder = new ViewHolder();
-
-        // set UI elements
-        holder.layout = view.findViewById(R.id.chatMessageLayout);
-
-        holder.tvMessage = view.findViewById(R.id.tvMessage);
-        holder.tvTimestamp = view.findViewById(R.id.tvTime);
-        if (outgoing) {
-            holder.pbSending = view.findViewById(R.id.progressBar);
-            holder.ivSent = view.findViewById(R.id.sentImage);
+        ChatMessage msg = getItem(position);
+        ViewHolder holder;
+        View listItem = convertView;
+        if (listItem == null)   {
+            listItem = LayoutInflater.from(mContext).inflate(layout, parent, false);
+            holder = new ViewHolder();
+            //Set UI elements
+            holder.layout = listItem.findViewById(R.id.chatMessageLayout);
+
+            holder.tvMessage = listItem.findViewById(R.id.tvMessage);
+            holder.tvTimestamp = listItem.findViewById(R.id.tvTime);
+            if(outgoing)    {
```

Comment:
that brace seems funky.
could you run the auto-formatter for all your committed files, please?